### ISSUE TITLE
ssh keyphrase and PEM keys not supported

### ISSUE THREAD

**Issue by buggedcom** [TablePlus/TablePlus#880]
1. Which driver are you using and version of it (Ex: PostgreSQL 10.0): MySQL 8

2. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81): 173

3. The steps to reproduce this issue:
![screenshot 2019-01-09 at 12 29 06](https://user-images.githubusercontent.com/2810/50894334-a6f85880-140b-11e9-8166-c2b0c8281634.png)

Try to connect via ssh alias that requires a keyphrase and PEM key cannot be done unless running via the command line using the TablePlus-Debug version, as there are no prompt's for the keys, it simply says 

![screenshot 2019-01-09 at 12 31 28](https://user-images.githubusercontent.com/2810/50893843-82e84780-140a-11e9-8cd2-a7ac4a7c189a.png)

**Comment by buggedcom** [comment]
However you cannot actually connect to the server once the connection is saved either.

```
Olivers-iMac-2:deleteme ollie$ ./TablePlus-Debug.app/Contents/MacOS/TablePlus
2019-01-09 12:33:18.169 TablePlus[20146:5914428] [Crashlytics] Version 3.12.0 (136)
2019-01-09 12:33:18.610 TablePlus[20146:5914538] -[NSURL init] called; this results in an NSURL instance with an empty URL string. Please use one of the documented NSURL initialization methods instead (initWithString:, initFileURLWithPath:, etc.). This message shown only once.
[2019/01/09 12:33:19.146670, 2] ssh_connect:  libssh 0.8.5 (c) 2003-2018 Aris Adamantiadis, Andreas Schneider and libssh contributors. Distributed under the LGPL, please refer to COPYING file for information about your rights, using threading threads_pthread
[2019/01/09 12:33:19.146942, 3] getai:  host 192.168.33.10 matches an IP address
[2019/01/09 12:33:19.147436, 2] ssh_socket_connect:  Nonblocking connection socket: 20
[2019/01/09 12:33:19.147462, 2] ssh_connect:  Socket connecting, now waiting for the callbacks to work
[2019/01/09 12:33:19.147471, 3] ssh_connect:  Actual timeout : 35000
[2019/01/09 12:33:19.149933, 4] ssh_socket_pollcallback:  Poll callback on socket 20 (POLLOUT ), out buffer 0
[2019/01/09 12:33:19.151610, 3] ssh_socket_pollcallback:  Received POLLOUT in connecting state
[2019/01/09 12:33:19.151695, 1] socket_callback_connected:  Socket connection callback: 1 (0)
[2019/01/09 12:33:19.152255, 3] ssh_socket_unbuffered_write:  Enabling POLLOUT for socket
[2019/01/09 12:33:19.152690, 4] ssh_socket_pollcallback:  Poll callback on socket 20 (POLLOUT ), out buffer 0
[2019/01/09 12:33:19.156734, 4] ssh_socket_pollcallback:  Poll callback on socket 20 (POLLIN ), out buffer 0
[2019/01/09 12:33:19.156968, 3] callback_receive_banner:  Received banner: SSH-2.0-OpenSSH_7.2p2 Ubuntu-4ubuntu2.2
[2019/01/09 12:33:19.157064, 1] ssh_client_connection_callback:  SSH server banner: SSH-2.0-OpenSSH_7.2p2 Ubuntu-4ubuntu2.2
[2019/01/09 12:33:19.157074, 1] ssh_analyze_banner:  Analyzing banner: SSH-2.0-OpenSSH_7.2p2 Ubuntu-4ubuntu2.2
[2019/01/09 12:33:19.157194, 1] ssh_analyze_banner:  We are talking to an OpenSSH client version: 7.2 (70200)
[2019/01/09 12:33:19.188963, 1] ssh_known_hosts_read_entries:  Failed to open the known_hosts file '/etc/ssh/ssh_known_hosts': No such file or directory
[2019/01/09 12:33:19.189015, 3] ssh_client_select_hostkeys:  Changing host key method to "ecdsa-sha2-nistp256,ssh-rsa,ssh-ed25519,ecdsa-sha2-nistp521,ecdsa-sha2-nistp384,ssh-dss"
[2019/01/09 12:33:19.189065, 4] ssh_list_kex:  kex algos: curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha1,diffie-hellman-group1-sha1,ext-info-c
[2019/01/09 12:33:19.189077, 4] ssh_list_kex:  server host key algo: ecdsa-sha2-nistp256,ssh-rsa,ssh-ed25519,ecdsa-sha2-nistp521,ecdsa-sha2-nistp384,ssh-dss
[2019/01/09 12:33:19.189085, 4] ssh_list_kex:  encryption client->server: aes256-ctr,aes192-ctr,aes128-ctr,aes256-cbc,aes192-cbc,aes128-cbc,blowfish-cbc,3des-cbc
[2019/01/09 12:33:19.189092, 4] ssh_list_kex:  encryption server->client: aes256-ctr,aes192-ctr,aes128-ctr,aes256-cbc,aes192-cbc,aes128-cbc,blowfish-cbc,3des-cbc
[2019/01/09 12:33:19.189098, 4] ssh_list_kex:  mac algo client->server: hmac-sha2-256,hmac-sha2-512,hmac-sha1
[2019/01/09 12:33:19.189105, 4] ssh_list_kex:  mac algo server->client: hmac-sha2-256,hmac-sha2-512,hmac-sha1
[2019/01/09 12:33:19.189112, 4] ssh_list_kex:  compression algo client->server: none
[2019/01/09 12:33:19.189118, 4] ssh_list_kex:  compression algo server->client: none
[2019/01/09 12:33:19.189125, 4] ssh_list_kex:  languages client->server: 
[2019/01/09 12:33:19.189131, 4] ssh_list_kex:  languages server->client: 
[2019/01/09 12:33:19.189487, 3] ssh_socket_unbuffered_write:  Enabling POLLOUT for socket
[2019/01/09 12:33:19.189941, 3] packet_send2:  packet: wrote [len=644,padding=9,comp=634,payload=634]
[2019/01/09 12:33:19.190403, 3] ssh_packet_socket_callback:  packet: read type 20 [len=972,padding=10,comp=961,payload=961]
[2019/01/09 12:33:19.190567, 3] ssh_packet_process:  Dispatching handler for packet type 20
[2019/01/09 12:33:19.190649, 4] ssh_list_kex:  kex algos: curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group14-sha1
[2019/01/09 12:33:19.190660, 4] ssh_list_kex:  server host key algo: ssh-rsa,rsa-sha2-512,rsa-sha2-256,ecdsa-sha2-nistp256,ssh-ed25519
[2019/01/09 12:33:19.190668, 4] ssh_list_kex:  encryption client->server: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com
[2019/01/09 12:33:19.190676, 4] ssh_list_kex:  encryption server->client: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com
[2019/01/09 12:33:19.190684, 4] ssh_list_kex:  mac algo client->server: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
[2019/01/09 12:33:19.190691, 4] ssh_list_kex:  mac algo server->client: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
[2019/01/09 12:33:19.190699, 4] ssh_list_kex:  compression algo client->server: none,zlib@openssh.com
[2019/01/09 12:33:19.190705, 4] ssh_list_kex:  compression algo server->client: none,zlib@openssh.com
[2019/01/09 12:33:19.190712, 4] ssh_list_kex:  languages client->server: 
[2019/01/09 12:33:19.190719, 4] ssh_list_kex:  languages server->client: 
[2019/01/09 12:33:19.190748, 2] ssh_kex_select_methods:  Negotiated curve25519-sha256@libssh.org,ecdsa-sha2-nistp256,aes256-ctr,aes256-ctr,hmac-sha2-256,hmac-sha2-256,none,none,,
[2019/01/09 12:33:19.208356, 3] packet_send2:  packet: wrote [len=44,padding=6,comp=37,payload=37]
[2019/01/09 12:33:19.208401, 4] ssh_socket_pollcallback:  Poll callback on socket 20 (POLLOUT ), out buffer 48
[2019/01/09 12:33:19.208425, 3] ssh_socket_unbuffered_write:  Enabling POLLOUT for socket
[2019/01/09 12:33:19.208444, 4] ssh_socket_pollcallback:  Poll callback on socket 20 (POLLOUT ), out buffer 0
[2019/01/09 12:33:19.208453, 4] ssh_socket_pollcallback:  sending control flow event
[2019/01/09 12:33:19.208460, 4] ssh_packet_socket_controlflow_callback:  sending channel_write_wontblock callback
[2019/01/09 12:33:19.242971, 4] ssh_socket_pollcallback:  Poll callback on socket 20 (POLLIN ), out buffer 0
[2019/01/09 12:33:19.243022, 3] ssh_packet_socket_callback:  packet: read type 31 [len=260,padding=11,comp=248,payload=248]
[2019/01/09 12:33:19.243035, 3] ssh_packet_process:  Dispatching handler for packet type 31
[2019/01/09 12:33:19.243044, 2] ssh_packet_dh_reply:  Received SSH_KEXDH_REPLY
[2019/01/09 12:33:19.253542, 3] ssh_socket_unbuffered_write:  Enabling POLLOUT for socket
[2019/01/09 12:33:19.253568, 3] packet_send2:  packet: wrote [len=12,padding=10,comp=1,payload=1]
[2019/01/09 12:33:19.253579, 2] ssh_client_curve25519_reply:  SSH_MSG_NEWKEYS sent
[2019/01/09 12:33:19.253588, 3] ssh_packet_socket_callback:  Processing 112 bytes left in socket buffer
[2019/01/09 12:33:19.253598, 3] ssh_packet_socket_callback:  packet: read type 21 [len=12,padding=10,comp=1,payload=1]
[2019/01/09 12:33:19.253606, 3] ssh_packet_process:  Dispatching handler for packet type 21
[2019/01/09 12:33:19.253615, 2] ssh_packet_newkeys:  Received SSH_MSG_NEWKEYS
[2019/01/09 12:33:19.254657, 3] crypt_set_algorithms2:  Set output algorithm to aes256-ctr
[2019/01/09 12:33:19.254674, 3] crypt_set_algorithms2:  Set HMAC output algorithm to hmac-sha2-256
[2019/01/09 12:33:19.254683, 3] crypt_set_algorithms2:  Set input algorithm to aes256-ctr
[2019/01/09 12:33:19.254690, 3] crypt_set_algorithms2:  Set HMAC input algorithm to hmac-sha2-256
[2019/01/09 12:33:19.255757, 4] ssh_pki_signature_verify_blob:  Going to verify a ssh-ecdsa type signature
[2019/01/09 12:33:19.258300, 2] ssh_packet_newkeys:  Signature verified and valid
[2019/01/09 12:33:19.258374, 3] ssh_packet_socket_callback:  Processing 96 bytes left in socket buffer
[2019/01/09 12:33:19.259413, 3] ssh_packet_socket_callback:  packet: read type 7 [len=60,padding=6,comp=53,payload=53]
[2019/01/09 12:33:19.259439, 3] ssh_packet_process:  Dispatching handler for packet type 7
[2019/01/09 12:33:19.259450, 3] ssh_packet_ext_info:  Received SSH_MSG_EXT_INFO
[2019/01/09 12:33:19.259466, 3] ssh_packet_ext_info:  Follows 1 extensions
[2019/01/09 12:33:19.259484, 3] ssh_packet_ext_info:  Extension: server-sig-algs=<rsa-sha2-256,rsa-sha2-512>
[2019/01/09 12:33:19.259496, 3] ssh_connect:  current state : 7
[2019/01/09 12:33:19.259809, 3] packet_send2:  packet: wrote [len=28,padding=10,comp=17,payload=17]
[2019/01/09 12:33:19.259827, 3] ssh_service_request:  Sent SSH_MSG_SERVICE_REQUEST (service ssh-userauth)
[2019/01/09 12:33:19.259873, 4] ssh_socket_pollcallback:  Poll callback on socket 20 (POLLOUT ), out buffer 64
[2019/01/09 12:33:19.259928, 3] ssh_socket_unbuffered_write:  Enabling POLLOUT for socket
[2019/01/09 12:33:19.260136, 4] ssh_socket_pollcallback:  Poll callback on socket 20 (POLLOUT ), out buffer 0
[2019/01/09 12:33:19.260149, 4] ssh_socket_pollcallback:  sending control flow event
[2019/01/09 12:33:19.260157, 4] ssh_packet_socket_controlflow_callback:  sending channel_write_wontblock callback
[2019/01/09 12:33:19.295551, 4] ssh_socket_pollcallback:  Poll callback on socket 20 (POLLIN ), out buffer 0
[2019/01/09 12:33:19.295673, 3] ssh_packet_socket_callback:  packet: read type 6 [len=28,padding=10,comp=17,payload=17]
[2019/01/09 12:33:19.295687, 3] ssh_packet_process:  Dispatching handler for packet type 6
[2019/01/09 12:33:19.295698, 3] ssh_packet_service_accept:  Received SSH_MSG_SERVICE_ACCEPT
[2019/01/09 12:33:19.296056, 3] ssh_socket_unbuffered_write:  Enabling POLLOUT for socket
[2019/01/09 12:33:19.296118, 3] packet_send2:  packet: wrote [len=60,padding=5,comp=54,payload=54]
[2019/01/09 12:33:19.296530, 4] ssh_socket_pollcallback:  Poll callback on socket 20 (POLLOUT ), out buffer 0
[2019/01/09 12:33:19.296572, 4] ssh_socket_pollcallback:  sending control flow event
[2019/01/09 12:33:19.296598, 4] ssh_packet_socket_controlflow_callback:  sending channel_write_wontblock callback
[2019/01/09 12:33:19.337548, 4] ssh_socket_pollcallback:  Poll callback on socket 20 (POLLIN ), out buffer 0
[2019/01/09 12:33:19.337604, 3] ssh_packet_socket_callback:  packet: read type 52 [len=12,padding=10,comp=1,payload=1]
[2019/01/09 12:33:19.337616, 3] ssh_packet_process:  Dispatching handler for packet type 52
[2019/01/09 12:33:19.338114, 3] ssh_packet_userauth_success:  Authentication successful
[2019/01/09 12:33:19.338139, 4] ssh_packet_userauth_success:  Received SSH_USERAUTH_SUCCESS
[2019/01/09 12:33:25.456268, 3] ssh_config_parse_file:  Reading configuration data from /Users/ollie/.ssh/config
[2019/01/09 12:33:25.456449, 3] ssh_config_parse_file:  Reading configuration data from /etc/ssh/ssh_config
[2019/01/09 12:33:25.456497, 2] ssh_config_parse_line:  Unapplicable option: SendEnv, line: 49

[2019/01/09 12:33:25.456524, 2] ssh_connect:  libssh 0.8.5 (c) 2003-2018 Aris Adamantiadis, Andreas Schneider and libssh contributors. Distributed under the LGPL, please refer to COPYING file for information about your rights, using threading threads_pthread
[2019/01/09 12:33:25.456562, 2] ssh_socket_connect_proxycommand:  Executing proxycommand 'ssh -p 2200 xxx@xxxxx nc db3 2200'
[2019/01/09 12:33:25.459465, 2] ssh_socket_connect_proxycommand:  ProxyCommand connection pipe: [21,22]
[2019/01/09 12:33:25.459574, 1] socket_callback_connected:  Socket connection callback: 1 (0)
[2019/01/09 12:33:25.459649, 3] ssh_socket_unbuffered_write:  Enabling POLLOUT for socket
[2019/01/09 12:33:25.459665, 2] ssh_connect:  Socket connecting, now waiting for the callbacks to work
[2019/01/09 12:33:25.459675, 3] ssh_connect:  Actual timeout : 35000
[2019/01/09 12:33:25.459735, 4] ssh_socket_pollcallback:  Poll callback on socket 22 (POLLOUT ), out buffer 0
Enter passphrase for key '/Users/ollie/.ssh/id_rsa': 
[2019/01/09 12:33:34.300494, 4] ssh_socket_pollcallback:  Poll callback on socket 22 (POLLIN ), out buffer 0
[2019/01/09 12:33:34.300582, 3] callback_receive_banner:  Received banner: SSH-2.0-OpenSSH_7.2p2 Ubuntu-4ubuntu2.6
[2019/01/09 12:33:34.300591, 1] ssh_client_connection_callback:  SSH server banner: SSH-2.0-OpenSSH_7.2p2 Ubuntu-4ubuntu2.6
[2019/01/09 12:33:34.300596, 1] ssh_analyze_banner:  Analyzing banner: SSH-2.0-OpenSSH_7.2p2 Ubuntu-4ubuntu2.6
[2019/01/09 12:33:34.300601, 1] ssh_analyze_banner:  We are talking to an OpenSSH client version: 7.2 (70200)
[2019/01/09 12:33:34.300854, 1] ssh_known_hosts_read_entries:  Failed to open the known_hosts file '/etc/ssh/ssh_known_hosts': No such file or directory
[2019/01/09 12:33:34.300875, 4] ssh_list_kex:  kex algos: curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha1,diffie-hellman-group1-sha1,ext-info-c
[2019/01/09 12:33:34.300882, 4] ssh_list_kex:  server host key algo: ssh-ed25519,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,ssh-rsa,rsa-sha2-512,rsa-sha2-256,ssh-dss
[2019/01/09 12:33:34.300890, 4] ssh_list_kex:  encryption client->server: aes256-ctr,aes192-ctr,aes128-ctr,aes256-cbc,aes192-cbc,aes128-cbc,blowfish-cbc,3des-cbc
[2019/01/09 12:33:34.300897, 4] ssh_list_kex:  encryption server->client: aes256-ctr,aes192-ctr,aes128-ctr,aes256-cbc,aes192-cbc,aes128-cbc,blowfish-cbc,3des-cbc
[2019/01/09 12:33:34.300905, 4] ssh_list_kex:  mac algo client->server: hmac-sha2-256,hmac-sha2-512,hmac-sha1
[2019/01/09 12:33:34.300924, 4] ssh_list_kex:  mac algo server->client: hmac-sha2-256,hmac-sha2-512,hmac-sha1
[2019/01/09 12:33:34.300933, 4] ssh_list_kex:  compression algo client->server: none
[2019/01/09 12:33:34.300938, 4] ssh_list_kex:  compression algo server->client: none
[2019/01/09 12:33:34.300944, 4] ssh_list_kex:  languages client->server: 
[2019/01/09 12:33:34.300952, 4] ssh_list_kex:  languages server->client: 
[2019/01/09 12:33:34.300981, 3] ssh_socket_unbuffered_write:  Enabling POLLOUT for socket
[2019/01/09 12:33:34.300992, 3] packet_send2:  packet: wrote [len=668,padding=7,comp=660,payload=660]
[2019/01/09 12:33:34.301018, 4] ssh_socket_pollcallback:  Poll callback on socket 22 (POLLOUT ), out buffer 0
[2019/01/09 12:33:34.301028, 4] ssh_socket_pollcallback:  sending control flow event
[2019/01/09 12:33:34.301035, 4] ssh_packet_socket_controlflow_callback:  sending channel_write_wontblock callback
[2019/01/09 12:33:34.336955, 4] ssh_socket_pollcallback:  Poll callback on socket 22 (POLLIN ), out buffer 0
[2019/01/09 12:33:34.337007, 3] ssh_packet_socket_callback:  packet: read type 20 [len=972,padding=10,comp=961,payload=961]
[2019/01/09 12:33:34.337022, 3] ssh_packet_process:  Dispatching handler for packet type 20
[2019/01/09 12:33:34.337045, 4] ssh_list_kex:  kex algos: curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group14-sha1
[2019/01/09 12:33:34.337054, 4] ssh_list_kex:  server host key algo: ssh-rsa,rsa-sha2-512,rsa-sha2-256,ecdsa-sha2-nistp256,ssh-ed25519
[2019/01/09 12:33:34.337062, 4] ssh_list_kex:  encryption client->server: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com
[2019/01/09 12:33:34.337070, 4] ssh_list_kex:  encryption server->client: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com
[2019/01/09 12:33:34.337078, 4] ssh_list_kex:  mac algo client->server: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
[2019/01/09 12:33:34.337122, 4] ssh_list_kex:  mac algo server->client: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
[2019/01/09 12:33:34.337134, 4] ssh_list_kex:  compression algo client->server: none,zlib@openssh.com
[2019/01/09 12:33:34.337142, 4] ssh_list_kex:  compression algo server->client: none,zlib@openssh.com
[2019/01/09 12:33:34.337197, 4] ssh_list_kex:  languages client->server: 
[2019/01/09 12:33:34.337207, 4] ssh_list_kex:  languages server->client: 
[2019/01/09 12:33:34.337271, 2] ssh_kex_select_methods:  Negotiated curve25519-sha256@libssh.org,ssh-ed25519,aes256-ctr,aes256-ctr,hmac-sha2-256,hmac-sha2-256,none,none,,
[2019/01/09 12:33:34.347788, 3] ssh_socket_unbuffered_write:  Enabling POLLOUT for socket
[2019/01/09 12:33:34.347847, 3] packet_send2:  packet: wrote [len=44,padding=6,comp=37,payload=37]
[2019/01/09 12:33:34.348025, 4] ssh_socket_pollcallback:  Poll callback on socket 22 (POLLOUT ), out buffer 0
[2019/01/09 12:33:34.348038, 4] ssh_socket_pollcallback:  sending control flow event
[2019/01/09 12:33:34.348046, 4] ssh_packet_socket_controlflow_callback:  sending channel_write_wontblock callback
[2019/01/09 12:33:34.404489, 4] ssh_socket_pollcallback:  Poll callback on socket 22 (POLLIN ), out buffer 0
[2019/01/09 12:33:34.404520, 3] ssh_packet_socket_callback:  packet: read type 31 [len=188,padding=8,comp=179,payload=179]
[2019/01/09 12:33:34.405024, 3] ssh_packet_process:  Dispatching handler for packet type 31
[2019/01/09 12:33:34.405039, 2] ssh_packet_dh_reply:  Received SSH_KEXDH_REPLY
[2019/01/09 12:33:34.439322, 3] ssh_socket_unbuffered_write:  Enabling POLLOUT for socket
[2019/01/09 12:33:34.439491, 3] packet_send2:  packet: wrote [len=12,padding=10,comp=1,payload=1]
[2019/01/09 12:33:34.439502, 2] ssh_client_curve25519_reply:  SSH_MSG_NEWKEYS sent
[2019/01/09 12:33:34.439510, 3] ssh_packet_socket_callback:  Processing 112 bytes left in socket buffer
[2019/01/09 12:33:34.439520, 3] ssh_packet_socket_callback:  packet: read type 21 [len=12,padding=10,comp=1,payload=1]
[2019/01/09 12:33:34.439558, 3] ssh_packet_process:  Dispatching handler for packet type 21
[2019/01/09 12:33:34.439571, 2] ssh_packet_newkeys:  Received SSH_MSG_NEWKEYS
[2019/01/09 12:33:34.439611, 3] crypt_set_algorithms2:  Set output algorithm to aes256-ctr
[2019/01/09 12:33:34.439622, 3] crypt_set_algorithms2:  Set HMAC output algorithm to hmac-sha2-256
[2019/01/09 12:33:34.439631, 3] crypt_set_algorithms2:  Set input algorithm to aes256-ctr
[2019/01/09 12:33:34.439639, 3] crypt_set_algorithms2:  Set HMAC input algorithm to hmac-sha2-256
[2019/01/09 12:33:34.439665, 4] ssh_pki_signature_verify_blob:  Going to verify a ssh-ed25519 type signature
[2019/01/09 12:33:34.489117, 2] ssh_packet_newkeys:  Signature verified and valid
[2019/01/09 12:33:34.489213, 3] ssh_packet_socket_callback:  Processing 96 bytes left in socket buffer
[2019/01/09 12:33:34.489254, 3] ssh_packet_socket_callback:  packet: read type 7 [len=60,padding=6,comp=53,payload=53]
[2019/01/09 12:33:34.489264, 3] ssh_packet_process:  Dispatching handler for packet type 7
[2019/01/09 12:33:34.489272, 3] ssh_packet_ext_info:  Received SSH_MSG_EXT_INFO
[2019/01/09 12:33:34.489279, 3] ssh_packet_ext_info:  Follows 1 extensions
[2019/01/09 12:33:34.489287, 3] ssh_packet_ext_info:  Extension: server-sig-algs=<rsa-sha2-256,rsa-sha2-512>
[2019/01/09 12:33:34.489298, 3] ssh_connect:  current state : 7
[2019/01/09 12:33:34.489500, 4] agent_talk:  Request length: 1
[2019/01/09 12:33:34.489887, 4] agent_talk:  Response length: 5
[2019/01/09 12:33:34.490154, 1] ssh_agent_get_ident_count:  Answer type: 12, expected answer: 12
[2019/01/09 12:33:34.490170, 3] ssh_agent_get_ident_count:  Agent count: 0
[2019/01/09 12:33:34.490178, 3] ssh_userauth_publickey_auto:  Trying to authenticate with /Users/ollie/.ssh/id_ed25519
[2019/01/09 12:33:34.490198, 1] ssh_pki_import_pubkey_file:  Error opening /Users/ollie/.ssh/id_ed25519.pub: No such file or directory
[2019/01/09 12:33:34.490212, 1] ssh_pki_import_privkey_file:  Error opening /Users/ollie/.ssh/id_ed25519: No such file or directory
[2019/01/09 12:33:34.490222, 3] ssh_userauth_publickey_auto:  Private key /Users/ollie/.ssh/id_ed25519 doesn't exist.
[2019/01/09 12:33:34.490230, 3] ssh_userauth_publickey_auto:  Trying to authenticate with /Users/ollie/.ssh/id_ecdsa
[2019/01/09 12:33:34.490244, 1] ssh_pki_import_pubkey_file:  Error opening /Users/ollie/.ssh/id_ecdsa.pub: No such file or directory
[2019/01/09 12:33:34.490257, 1] ssh_pki_import_privkey_file:  Error opening /Users/ollie/.ssh/id_ecdsa: No such file or directory
[2019/01/09 12:33:34.490265, 3] ssh_userauth_publickey_auto:  Private key /Users/ollie/.ssh/id_ecdsa doesn't exist.
[2019/01/09 12:33:34.490272, 3] ssh_userauth_publickey_auto:  Trying to authenticate with /Users/ollie/.ssh/id_rsa
[2019/01/09 12:33:34.490392, 3] ssh_key_algorithm_allowed:  Checking rsa-sha2-512 with list <ssh-ed25519,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,ssh-rsa,rsa-sha2-512,rsa-sha2-256,ssh-dss>
[2019/01/09 12:33:34.490474, 3] ssh_key_algorithm_allowed:  Checking rsa-sha2-512 with list <ssh-ed25519,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,ssh-rsa,rsa-sha2-512,rsa-sha2-256,ssh-dss>
[2019/01/09 12:33:34.491016, 3] packet_send2:  packet: wrote [len=28,padding=10,comp=17,payload=17]
[2019/01/09 12:33:34.491057, 3] ssh_service_request:  Sent SSH_MSG_SERVICE_REQUEST (service ssh-userauth)
[2019/01/09 12:33:34.491162, 4] ssh_socket_pollcallback:  Poll callback on socket 22 (POLLOUT ), out buffer 64
[2019/01/09 12:33:34.491188, 3] ssh_socket_unbuffered_write:  Enabling POLLOUT for socket
[2019/01/09 12:33:34.491353, 4] ssh_socket_pollcallback:  Poll callback on socket 22 (POLLOUT ), out buffer 0
[2019/01/09 12:33:34.491381, 4] ssh_socket_pollcallback:  sending control flow event
[2019/01/09 12:33:34.491392, 4] ssh_packet_socket_controlflow_callback:  sending channel_write_wontblock callback
[2019/01/09 12:33:34.537056, 4] ssh_socket_pollcallback:  Poll callback on socket 22 (POLLIN ), out buffer 0
[2019/01/09 12:33:34.537107, 3] ssh_packet_socket_callback:  packet: read type 6 [len=28,padding=10,comp=17,payload=17]
[2019/01/09 12:33:34.537118, 3] ssh_packet_process:  Dispatching handler for packet type 6
[2019/01/09 12:33:34.537127, 3] ssh_packet_service_accept:  Received SSH_MSG_SERVICE_ACCEPT
[2019/01/09 12:33:34.537235, 3] ssh_socket_unbuffered_write:  Enabling POLLOUT for socket
[2019/01/09 12:33:34.537269, 3] packet_send2:  packet: wrote [len=604,padding=6,comp=597,payload=597]
[2019/01/09 12:33:34.537292, 4] ssh_socket_pollcallback:  Poll callback on socket 22 (POLLOUT ), out buffer 0
[2019/01/09 12:33:34.537301, 4] ssh_socket_pollcallback:  sending control flow event
[2019/01/09 12:33:34.537308, 4] ssh_packet_socket_controlflow_callback:  sending channel_write_wontblock callback
[2019/01/09 12:33:34.588167, 4] ssh_socket_pollcallback:  Poll callback on socket 22 (POLLIN ), out buffer 0
[2019/01/09 12:33:34.588229, 3] ssh_packet_socket_callback:  packet: read type 60 [len=572,padding=15,comp=556,payload=556]
[2019/01/09 12:33:34.588240, 3] ssh_packet_process:  Dispatching handler for packet type 60
[2019/01/09 12:33:34.588249, 4] ssh_packet_userauth_pk_ok:  Received SSH_USERAUTH_PK_OK/INFO_REQUEST/GSSAPI_RESPONSE
[2019/01/09 12:33:34.588257, 4] ssh_packet_userauth_pk_ok:  Assuming SSH_USERAUTH_PK_OK
[2019/01/09 12:33:34.588411, 2] ssh_pki_import_privkey_base64:  Trying to decode privkey passphrase=false
Enter PEM pass phrase:
[2019/01/09 12:33:36.037590, 3] ssh_key_algorithm_allowed:  Checking rsa-sha2-512 with list <ssh-ed25519,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,ssh-rsa,rsa-sha2-512,rsa-sha2-256,ssh-dss>
[2019/01/09 12:33:36.037954, 3] ssh_key_algorithm_allowed:  Checking rsa-sha2-512 with list <ssh-ed25519,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,ssh-rsa,rsa-sha2-512,rsa-sha2-256,ssh-dss>
[2019/01/09 12:33:36.039776, 3] ssh_key_algorithm_allowed:  Checking rsa-sha2-512 with list <ssh-ed25519,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,ssh-rsa,rsa-sha2-512,rsa-sha2-256,ssh-dss>
[2019/01/09 12:33:36.058573, 3] ssh_socket_unbuffered_write:  Enabling POLLOUT for socket
[2019/01/09 12:33:36.058594, 3] packet_send2:  packet: wrote [len=1148,padding=14,comp=1133,payload=1133]
[2019/01/09 12:33:36.058619, 4] ssh_socket_pollcallback:  Poll callback on socket 22 (POLLOUT ), out buffer 0
[2019/01/09 12:33:36.058628, 4] ssh_socket_pollcallback:  sending control flow event
[2019/01/09 12:33:36.058637, 4] ssh_packet_socket_controlflow_callback:  sending channel_write_wontblock callback
[2019/01/09 12:33:36.136887, 4] ssh_socket_pollcallback:  Poll callback on socket 22 (POLLIN ), out buffer 0
[2019/01/09 12:33:36.136949, 3] ssh_packet_socket_callback:  packet: read type 52 [len=12,padding=10,comp=1,payload=1]
[2019/01/09 12:33:36.136960, 3] ssh_packet_process:  Dispatching handler for packet type 52
[2019/01/09 12:33:36.136969, 3] ssh_packet_userauth_success:  Authentication successful
[2019/01/09 12:33:36.136977, 4] ssh_packet_userauth_success:  Received SSH_USERAUTH_SUCCESS
[2019/01/09 12:33:36.137081, 2] ssh_userauth_publickey_auto:  Successfully authenticated using /Users/ollie/.ssh/id_rsa

```

**Comment by huyphams** [comment]
Wow, interesting @buggedcom I don't know why those key pem request key phrase this way. Maybe it's the root of the problem of many users out there.
Can I have a dummy key that has the same behavior with your key? (I mean it requests keyphrase in terminal).

**Comment by ashishwadekar** [comment]
Hi @huyphams 

Facing similar issue for all my connections with SSH using PEM key
<img width="965" alt="screenshot 2019-01-11 at 8 38 32 am" src="https://user-images.githubusercontent.com/13130213/51019817-b945d400-157c-11e9-9141-9ec03ef37149.png">

Let me know if a new Issue needs to be opened.

Cheers,
Ashish

**Comment by huyphams** [comment]
Hi @ashishwadekar  it looks like an PostgreSQL issue (not related to SSH issues), please follow the answer here: https://stackoverflow.com/questions/23375740/pgconnectionbad-fe-sendauth-no-password-supplied

**Comment by huyphams** [comment]
Hi @buggedcom this build should fix your issue: https://www.dropbox.com/s/0i0ofnhi88vlxz8/TablePlus.dmg?dl=0
Thanks again for reporting the issue.

**Comment by huyphams** [comment]
I've fixed it on the release 176, thanks everybody.
